docs/source/install/rhel7.rst

```diff
@@ -0,0 +1,387 @@
+
+At this point you have a minimal working installation, and can happily play with StackStorm:
+follow :doc:`/start` tutorial, :ref:`deploy examples <start-deploy-examples>`, explore and install packs from `st2contrib`_.
+
+But there is no joy without WebUI, no security without SSL termination, no fun without ChatOps, and no money without Enterprise edition. Read on, move on!
+
+-----------------
+
+Configure Authentication
+------------------------
+
+Reference deployment uses File Based auth provider for simplicity. Refer to :doc:`/authentication` to configure and use PAM or LDAP autentication backends. To set up authentication with File Based provider:
+
+* Create a user with a password:
+
+  .. code-block:: bash
+
+    # Install htpasswd utility if you don't have it
+    sudo yum -y install httpd-tools
+    # Create a user record in a password file.
+    echo "Ch@ngeMe" | sudo htpasswd -i /etc/st2/htpasswd test
+
+* Enable and configure auth in ``/etc/st2/st2.conf``:
+
+  .. sourcecode:: ini
+
+    [auth]
+    # ...
+    enabled = True
+    backend = flat_file
+    backend_kwargs = {"file_path": "/etc/st2/htpasswd"}
+    # ...
+
+* Restart the st2api service: ::
+
+    sudo st2ctl restart-component st2api
+
+* Authenticate, export the token for st2 CLI, and check that it works:
+
+  .. code-block:: bash
+
+    # Get an auth token and use in CLI or API
+    st2 auth test
+
+    # A shortcut to authenticate and export the token
+    export ST2_AUTH_TOKEN=$(st2 auth test -p Ch@ngeMe -t)
+
+    # Check that it works
+    st2 action list
+
+Check out :doc:`/cli` to learn convinient ways to authenticate via CLI.
+
+Install WebUI and setup SSL termination
+---------------------------------------
+`NGINX <http://nginx.org/>`_ is used to serve WebUI static files, redirect HTTP to HTTPS,
+provide SSL termination for HTTPS, and reverse-proxy st2auth and st2api API endpoints.
+To set it up: install `st2web` and `nginx`, generate certificates or place your existing
+certificates under ``/etc/ssl/st2``, and configure nginx with StackStorm's supplied
+:github_st2:`site config file st2.conf<conf/nginx/st2.conf>`.
+
+  .. code-block:: bash
+
+    # Install st2web and nginx
+    sudo yum install -y st2web nginx
+
+    # Generate self-signed certificate or place your existing certificate under /etc/ssl/st2
+    sudo mkdir -p /etc/ssl/st2
+    sudo openssl req -x509 -newkey rsa:2048 -keyout /etc/ssl/st2/st2.key -out /etc/ssl/st2/st2.crt \
+    -days 365 -nodes -subj "/C=US/ST=California/L=Palo Alto/O=StackStorm/OU=Information \
+    Technology/CN=$(hostname)"
+
+    # Copy and enable StackStorm's supplied config file
+    sudo cp /usr/share/doc/st2/conf/nginx/st2.conf /etc/nginx/conf.d/
+
+    # Disable default_server configuration in existing /etc/nginx/nginx.conf
+    sudo sed -i 's/default_server//g' /etc/nginx/nginx.conf
+
+    sudo systemctl restart nginx
+
+If you modify ports, or url paths in nginx configuration, make correspondent chagnes in st2web
```

Comment:
Problem on AWS Redhat 7, ami: RHEL-7.2_HVM_GA-20151112-x86_64-1-Hourly2-GP2 (ami-775e4f16)

### Symptoms: 
After restarting nginx web ui login comes up but the https://<host>/api and https://<host>/auth ports return 500 bad gateway. Checking /var/log/nginx/ssl-st2webui.error.log, there are errors: 
```
2016/02/27 22:44:59 [crit] 24354#0: *16 connect() to 127.0.0.1:9101 failed (13: Permission denied) while connecting to upstream, client: 69.181.118.28, server: , request: "GET /api/ HTTP/1.1", upstream: "http://127.0.0.1:9101/", host: "52.36.162.164"
2016/02/27 22:45:09 [crit] 24354#0: *16 connect() to 127.0.0.1:9100 failed (13: Permission denied) while connecting to upstream, client: 69.181.118.28, server: , request: "GET /auth/ HTTP/1.1", upstream: "http://127.0.0.1:9100/", host: "52.36.162.164"
```

Apparently, SELinux problem, common on Fedora/CentOS/RHEL (see https://stackoverflow.com/questions/23948527/13-permission-denied-while-connecting-to-upstreamnginx) 

`setsebool -P httpd_can_network_connect 1` fixed the problem. May be a more refined policy described in stackoverflow may be a more proper solution, but I'd say, up to a user now. 

It may be better to add "Troubleshooting" section and add references there, as we don't clearly know yet if it's a distro or image problem.